### PR TITLE
Resolve #1504, #1374, #1525, #1528, #1527, #1538, #1537

### LINKED ISSUE #1504: Map emailSendFailure to SEND_FAILURE in transfer state
This enables the transfer state to be shown as send failed in transfer history

Also modify corresponding receipt notification message for this situation to be

```
This transfer was not successful likely due to an invalid email address of {{receiverName}}.
You need to sign in to cancel the transfer first, then update the email address for {{receiverName}} and make a new transfer.
```

### PR DESCRIPTION
**Pull request by InVincible2016**

### CODE REVIEW

File: src/wallet.js [hunk 1]
```diff
@@ -109,16 +109,16 @@ function isReceivable ({ status, walletType }: { status: Object, walletType: str
 }
 
 function isAddable ({ status, walletType }: { status: Object, walletType: string }) {
+  if (['drive', 'metamaskOne', 'referralWallet', 'escrow'].includes(walletType)) {
+    status.addable = false
+    status.disabledReason = status.disabledReason || 'Not addable'
```
Comment (timothywangdev) [inline src/wallet.js:114]: do not include disablereason for internal wallets

Comment (timothywangdev) [reply src/wallet.js:114]: our wallet disabled reason is not specific enough, causing some bugs such as when sending from coinbase wallet,
"not supported on mobile browser" is shown.

Suggest introducing [send|receive|add]DisabledReason

File: src/components/LoginComponent.jsx [hunk 2]
```diff
@@ -227,17 +227,37 @@ class LoginComponent extends Component {
...
+                    >
+                      Term of Service
+                    </Link>{' '}
+                    and{' '}
+                    <Link
+                      className={classes.linkText}
+                      variant='caption'
+                      align='center'
+                      color='textSecondary'
+                      href={data.termURL}
```
Comment (timothywangdev) [inline src/components/LoginComponent.jsx:254]: should use privacy url

Comment (timothywangdev) [reply src/components/LoginComponent.jsx:254]: https://help.chainsfr.com/en/articles/3961208-privacy-policy

File: src/wallet.js (hunk 1 above)
Comment (InVincible2016) [reply src/wallet.js:114]: I think it's better to include disabledReason for internal wallets, simple logic, whenever a wallet is disabled, state the reason.
Will introduce [send|receive|add]DisabledReason

File: src/components/AccountDropdownComponent.jsx [hunk 3]
```diff
@@ -155,20 +155,27 @@ class AccountDropdownComponent extends Component<Props, State> {
...
+          <Box>
+            {/* name and wallet title*/}
+            <Typography variant='body2'>{item.name}</Typography>
+            <Typography variant='caption'>
+              {getWalletTitle(item.walletType)}
+              {item.platformType && `, ${getPlatformTitle(item.platformType)}`}
+            </Typography>
+            {!sendable && (
+              <Typography variant='caption' component='p'>
+                Sending not available on mobile browser
```
[the remark this answers is not in the document]
Comment (InVincible2016) [reply src/components/AccountDropdownComponent.jsx:175]: Will put 'Sending not available on mobile browser' in sendDisabledReason then

File: src/components/AddAccountModalComponent.jsx [hunk 4]
```diff
@@ -118,8 +120,207 @@ const useStyles = makeStyles(theme => ({
...
+
+type ViewAddressFeatureProps = {
+  supported: boolean,
+  walletType: string,
+  platformType: string
+}
+
+const ViewAddressFeature = (props: ViewAddressFeatureProps) => {
+  const { supported, walletType, platformType } = props
+  console.log('platformType', platformType)
```
Comment (timothywangdev) [inline src/components/AddAccountModalComponent.jsx:231]: remove

File: src/wallet.js [hunk 5]
```diff
@@ -90,11 +94,12 @@ function isSendable ({ status, walletType }: { status: Object, walletType: strin
     ].includes(walletType)
   ) {
     status.sendable = false
-    status.disabledReason = status.disabledReason || 'Not supported in mobile device'
+    status.sendDisabledReason = status.sendDisabledReason || 'Not supported in mobile device'
```
Comment (timothywangdev) [inline src/wallet.js:97]: coinbaseOauthWallet is supported on mobile device

**Pull request merged**